src/nupic/research/connections.py

```diff
@@ -243,6 +265,52 @@ def updateSynapsePermanence(self, synapse, permanence):
     self._synapsesForPresynapticCell[newData.presynapticCell][synapse] = newData
 
 
+  def computeActivity(self, activeInput, activePermanenceThreshold,
+                      activeSynapseThreshold, matchingPermananceThreshold,
+                      matchingSynapseThreshold):
+    """
+    Computes active and matching segments given the current active input.
```

Comment:
You can start doc strings on the first line:
https://www.python.org/dev/peps/pep-0257/